openstates/hi/people.py

```diff
@@ -0,0 +1,241 @@
+
+        chamber = chamber[0].text_content()
+        ret['chamber'] = chamber
+
+        if table is not None:
+            cttys = table.xpath("./tr/td/a")
+            for ctty in cttys:
+                ret['ctty'].append({
+                    "name": ctty.text,
+                    "page": "%s/%s" % (HI_BASE_URL, ctty.attrib['href']),
+                })
+        return ret
+
+    def scrape_leg_page(self, url):
+        page = self.get_page(url)
+        people = page.xpath(
+            "//table[@id='ctl00_ContentPlaceHolderCol1_GridView1']")[0]
+        people = people.xpath('./tr')[1:]
+        display_order = {
+            "image": 0,
+            "contact": 1,
+            "district": 2
+        }
+
+        ret = []
+
+        for person in people:
+            image = person[display_order["image"]]
+            contact = person[display_order["contact"]]
+            district = person[display_order["district"]]
+            metainf = self.scrape_contact_info(contact)
+            district = self.scrape_district_info(district)
+            homepage = self.scrape_homepage(metainf['homepage'])
+
+            image = "%s/%s" % (
+                HI_BASE_URL,
+                image.xpath("./*/img")[0].attrib['src']
+            )
+
+            pmeta = {
+                "image": image,
+                "source": [url],
+                "district": district,
+                "chamber": None
+            }
+
+            if homepage is not None:
+                pmeta['source'].append(homepage['source'])
+                pmeta['ctty'] = homepage['ctty']
+                pmeta['chamber'] = homepage['chamber']
+
+            if pmeta['chamber'] is None:
+                raise Exception("No chamber found.")
+
+            for meta in metainf:
+                pmeta[meta] = metainf[meta]
+
+            ret.append(pmeta)
+        return ret
+
+    def br_split(self, contact):
+        cel = []
+        els = [cel]
+
+        # krufty HTML requires stupid hacks
+        elements = contact.xpath("./*")
+        for element in elements:
+            if element.tag == "br":
+                cel = []
+                els.append(cel)
+            else:
+                cel.append(element)
+        return els
+
+    def scrape_district_info(self, district):
+        return district[2].text_content()
+
+    def scrape_contact_info(self, contact):
+        homepage = "%s/%s" % (
+            HI_BASE_URL,
+            contact.xpath("./a")[0].attrib['href']
+        )
+
+        els = self.br_split(contact)
+
+        def _scrape_title(els):
+            return els[0].text_content()
+
+        def _scrape_name(els):
+            lName = els[0].text_content()
+            fName = els[2].text_content()
+            return "%s %s" % (fName, lName)
+
+        def _scrape_party(els):
+            party = {
+                "(D)": "Democratic",
+                "(R)": "Republican"
+            }
+
+            try:
```

Comment:
Looks like this block can be simplified to `return party.get(els[4].text_content(), 'Other')`.